discussions-server/src/main/java/org/xwiki/contrib/discussions/internal/server/DefaultDiscussionMessageRequestCreator.java

```diff
@@ -0,0 +1,184 @@
+ */
+package org.xwiki.contrib.discussions.internal.server;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.discussions.DiscussionReferencesResolver;
+import org.xwiki.contrib.discussions.DiscussionService;
+import org.xwiki.contrib.discussions.DiscussionStoreConfigurationParameters;
+import org.xwiki.contrib.discussions.MessageService;
+import org.xwiki.contrib.discussions.domain.Discussion;
+import org.xwiki.contrib.discussions.domain.Message;
+import org.xwiki.contrib.discussions.domain.references.ActorReference;
+import org.xwiki.contrib.discussions.domain.references.DiscussionReference;
+import org.xwiki.contrib.discussions.domain.references.MessageReference;
+import org.xwiki.contrib.discussions.server.DiscussionMessageRequestCreator;
+import org.xwiki.contrib.discussions.server.DiscussionServerException;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.rendering.parser.ParseException;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.wysiwyg.converter.HTMLConverter;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.EditForm;
+
+import static org.xwiki.rendering.syntax.Syntax.XWIKI_2_0;
+
+/**
+ * Default implementation of {@link DiscussionMessageRequestCreator}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@Component
+@Singleton
+public class DefaultDiscussionMessageRequestCreator implements DiscussionMessageRequestCreator
+{
+    @Inject
+    private DiscussionService discussionService;
+
+    @Inject
+    private DiscussionReferencesResolver discussionReferencesResolver;
+
+    @Inject
+    private HTMLConverter htmlConverter;
+
+    @Inject
+    private MessageService messageService;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Override
+    public Optional<Message> createMessage(HttpServletRequest request) throws DiscussionServerException
+    {
+        String serializedReference = request.getParameter(DISCUSSION_REFERENCE_PARAM);
+        if (StringUtils.isBlank(serializedReference)) {
+            throw new DiscussionServerException(HttpServletResponse.SC_BAD_REQUEST,
+                "The discussion reference has not been provided.");
+        } else {
+            DiscussionReference discussionReference =
+                this.discussionReferencesResolver.resolve(serializedReference, DiscussionReference.class);
+            Optional<Discussion> discussionOptional = this.discussionService.get(discussionReference);
+            if (!discussionOptional.isPresent()) {
```

Comment:
[orElseThrow](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-)  is more idiomatic.